main/pipeline/pipeline.cc

```diff
@@ -572,7 +572,17 @@ ast::ParsedFile typecheckOne(core::Context ctx, ast::ParsedFile resolved, const
         }
         CFGCollectorAndTyper collector(opts);
         {
-            result.tree = ast::TreeMap::apply(ctx, collector, move(resolved.tree));
+            if (f.data(ctx).isPackage()) {
+#ifndef SORBET_REALMAIN_MIN
+                auto removedMods = packager::Packager::removePackageModules(ctx, resolved);
+                result.tree = ast::TreeMap::apply(ctx, collector, move(resolved.tree));
+                result = packager::Packager::replacePackageModules(ctx, move(result), move(removedMods));
+#else
+                ENFORCE(false, "Should never happen");
+#endif
+            } else {
+                result.tree = ast::TreeMap::apply(ctx, collector, move(resolved.tree));
```

Comment:
Heads up that I opened https://github.com/sorbet/sorbet/pull/4951 after seeing that we are doing a full treemap here and not a shallowmap. That will conflict with this PR.

I don't think that my change will fix the packager overhead, though -- ShallowMap doesn't recur into method bodies but still recurs into class bodies.